### PR TITLE
Update Ruby doc for latest 3.1 version

### PR DESCRIPTION
**Pull request by st0012**
Hello 👋, Ruby SDK's latest version [`3.1.0`](https://github.com/getsentry/raven-ruby/releases/tag/3.1.0) was just released. And this PR updates the documents to reflect the changes.

### CODE REVIEW

**Review by rhcarvalho — APPROVED** [review]
Excellent, thanks Stan! Will let the wordsmiths have a quick pass before we merge.

File: src/platforms/ruby/common/configuration/options.mdx [hunk 1]
```diff
@@ -40,6 +40,30 @@ end
 
 If the async callback raises an exception, Raven will attempt to send synchronously.
 
+`backtrace_cleanup_callback`
+
+: In case you want to clean up exceptions' backtrace before it's sent to Sentry, you can specify a callback with `backtrace_cleanup_callback` to do that. For example:
```
Comment (PeloWriter) [inline src/platforms/ruby/common/configuration/options.mdx:45]:
```suggestion
: If you want to clean up exceptions' backtrace before it's sent to Sentry, you can specify a callback with `backtrace_cleanup_callback` to do that. For example:
```

File: src/platforms/ruby/common/configuration/options.mdx [hunk 2]
```diff
@@ -40,6 +40,30 @@ end
...
+
+```ruby
+config.backtrace_cleanup_callback = lambda do |backtrace|
+  Rails.backtrace_cleaner.clean(backtrace)
+end
+```
+
+`breadcrumbs_logger`
+
+: Sentry supports 2 breadcrumbs loggers in the Ruby SDK:
```
Comment (PeloWriter) [inline src/platforms/ruby/common/configuration/options.mdx:55]:
```suggestion
: Sentry supports two breadcrumbs loggers in the Ruby SDK:
```

File: src/platforms/ruby/common/configuration/options.mdx [hunk 3]
```diff
@@ -40,6 +40,30 @@ end
...
+- `SentryLogger` - A general breadcrumbs logger for all Ruby applications.
+- `ActiveSupportLogger` - Built on top of Rails ActiveSupport's instrumentation and provide many Rails-specific information.
```
Comment (PeloWriter) [inline src/platforms/ruby/common/configuration/options.mdx:57]:
```suggestion
- `ActiveSupportLogger` - Built on top of Rails ActiveSupport's instrumentation and provides many Rails-specific information.
```

**Review by PeloWriter — APPROVED** [review]
A touch of smoothing, but otherwise LGTM. Thank you.

**Comment by st0012** [comment]
@PeloWriter all updated, thank you!

**Pull request merged**